mlflow/store/tracking/file_store.py

```diff
@@ -931,3 +970,91 @@ def record_logged_model(self, run_id, mlflow_model):
             self._set_run_tag(run_info, tag)
         except Exception as e:
             raise MlflowException(e, INTERNAL_ERROR)
+
+
+def _like_pattern_to_regex(pattern):
+    """
+    Convert a pattern string for LIKE and ILIKE to the equivalent regular expression.
+    """
+    return "^" + pattern.replace("%", ".*") + "$"
+
+
+def _create_matcher(comparator, value):
+    if comparator == "=":
+        return value.__eq__
+    elif comparator == "!=":
+        return value.__ne__
+    elif comparator in ("LIKE", "ILIKE"):
+        flag = re.IGNORECASE if comparator == "ILIKE" else 0
+        regex = re.compile(_like_pattern_to_regex(value), flag)
+        return regex.match
+    raise MlflowException.invalid_parameter_value(f"Invalid comparator: {comparator}")
+
+
+def filter_by_attribute(key, comparator, value):
+    matcher = _create_matcher(comparator, value)
+
+    def matched(experiment):
+        return matcher(getattr(experiment, key))
+
+    return matched
+
+
+def filter_by_tags(key, comparator, value):
+    matcher = _create_matcher(comparator, value)
+
+    def has_tag_and_matched(experiment):
+        val = experiment.tags.get(key)
+        if val is not None:
+            return matcher(val)
+        return False
+
+    return has_tag_and_matched
```

Comment:
A lot of similar logic seems to exist in https://github.com/mlflow/mlflow/blob/e1f493b8de58a2ebbc4d8a3ef27ee2166f748586/mlflow/utils/search_utils.py#L395-L435. Anything we can reuse from there instead?

Reply:
Updated the code to reuse the existing logic :)